### ISSUE TITLE
time: Parse of am/pm should be forgiving of spaces and case

### ISSUE THREAD

**Issue by spekary** [golang/go#51320]
### What version of Go are you using (`go version`)? 

<pre>
1.17.5
</pre>

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
Mac OS
</pre></details>

### What did you do?

import (
	"testing"
	"time"
)

func TestParse(t *testing.T) {
	var tm time.Time
	var err error

	tm, err = time.Parse("3:04PM", "5:23 PM")
	if err != nil {
		t.Error(err)
	}
	if tm.Hour() != 17 {
		t.Error("hour does not match")
	}

	tm, err = time.Parse("3:04pm", "5:23PM")
	if err != nil {
		t.Error(err)
	}
	if tm.Hour() != 17 {
		t.Error("hour does not match")
	}
}

### What did you expect to see?

Tests pass

### What did you see instead?

=== RUN   TestParse
    parse_test.go:14: parsing time "5:23 PM" as "3:04PM": cannot parse "M" as "PM"
    parse_test.go:17: hour does not match
    parse_test.go:22: parsing time "5:23PM" as "3:04pm": cannot parse "" as "pm"
    parse_test.go:25: hour does not match
--- FAIL: TestParse (0.00s)

**Comment by thanm** [comment]
@rsc per owners.

It seems as though it would be very easy to write a bit of Go code that would preprocess time values to put them in canonical form prior to the time.Parse() invocation. That is, if you want  both "PM" and "pm" to be accepted, just pattern match the tail end of the value string and replace it if needed.

**Comment by andlabs** [comment]
Another example, including periods and dots:

```
parsing time "07/13/2023 9:34 A.M." as "01/02/2006 3:04 P.M.": cannot parse "A.M." as " P.M."
```

(error message is self-explanatory; just tried this in the 1.20 playground)

**Comment by seankhliao** [comment]
I don't think this is something we can change, people generally expect more strictness from the parser.
